File: daq/slc/src/dqm/src/ShortArray.cc
#include "ShortArray.h"

#include <sstream>

using namespace Belle2;

ShortArray::ShortArray() throw() : NumberArray(0), _value_v(NULL) {}

ShortArray::ShortArray(size_t size) throw() : NumberArray(), _value_v(NULL)
{
  resize(size);
}

ShortArray::ShortArray(const NumberArray& v) throw() : NumberArray(), _value_v(NULL)
{
  resize(v.size());
  for (size_t n = 0; n < size(); n++) set(n, _value_v[n]);
}

ShortArray::~ShortArray() throw()
{
  if (_value_v != NULL) delete [] _value_v;
}

double ShortArray::get(size_t n) const throw()
{
  if (n < size()) return _value_v[n];
  else return -1;
}
void ShortArray::set(size_t n, double v) throw()
{
  if (n < size()) _value_v[n] = (short)v;
}
void ShortArray::resize(int s) throw()
{
  if (_value_v != NULL) delete [] _value_v;
  _value_v = new short [s];
  _size = s;
  for (size_t n = 0; n < size(); n++) set(n, 0);
}

std::string ShortArray::toScript() const throw()
{
  std::stringstream ss("");
  for (size_t n = 0; n < size(); n++) {
    ss << get(n);
    if (n < size() - 1) ss << ",";
  }
  return ss.str();
}

short& ShortArray::operator[](size_t n) throw()
{
  if (n < size()) return _value_v[n];
  return _value_v[0];
}

short ShortArray::operator[](size_t n) const throw()
{
  if (n < size()) return _value_v[n];
  return 0;
}

void ShortArray::readObject(Belle2::Reader& reader)
throw(Belle2::IOException)
{
  for (size_t n = 0; n < size(); n++) {
    _value_v[n] = reader.readShort();
  }
}

void ShortArray::writeObject(Belle2::Writer& writer)
const throw(Belle2::IOException)
{
  for (size_t n = 0; n < size(); n++) {
    writer.writeShort(_value_v[n]);
  }
}

double ShortArray::readElement(Belle2::Reader& reader)
throw(Belle2::IOException)
{
  return reader.readShort();
}

void ShortArray::writeElement(double value, Belle2::Writer& writer)
const throw(Belle2::IOException)
{
  writer.writeShort(value);
}

File: daq/slc/extra/javalib/b2daq/src/b2daq/dqm/core/ShortArray.java
package b2daq.dqm.core;

import java.io.IOException;

import b2daq.core.Reader;
import b2daq.core.Writer;

public class ShortArray extends NumberArray {

	private short[] _value_v;
	
	public ShortArray() {
		_value_v = new short [0];
		for (int n = 0; n < length(); n++ ) _value_v[n] = 0;
	}

	public ShortArray(int len) {
		_value_v = new short [len];
		for (int n = 0; n < length(); n++ ) _value_v[n] = 0;
	}

	public ShortArray(NumberArray array) { copy(array); }

	public void copy(short [] array) {
		_value_v = new short [array.length];
		for (int n = 0; n < length(); n++ ) _value_v[n] = (short)array[n];
	}

	public void copy(NumberArray array) {
		_value_v = new short [array.length()];
		for (int n = 0; n < length(); n++ ) _value_v[n] = (short)array.get(n);
	}

	public double get(int index) {
		if ( index >= 0 && index < length() ) return _value_v[index];
		else return -1;
	}

	public void set(int index, double data) {
		if ( index >= 0 && index < length() ) _value_v[index] = (short)data;
	}

	public void set(int index, short data) {
		if ( index >= 0 && index < length() ) _value_v[index] = data;
	}

	public int length() { return _value_v.length; }

	public void resize(int len) {
		_value_v = new short [len];
		for (int n = 0; n < length(); n++ ) _value_v[n] = 0;
	}

	public void writeSingle(double value, Writer writer) throws IOException {
		writer.writeShort((short)value);
	}

	public double readSingle(Reader reader) throws IOException {
		return reader.readShort();
	}

}
